pkg/registry/core/pod/storage/eviction.go

```diff
@@ -138,46 +138,47 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		return &metav1.Status{
 			Status: metav1.StatusSuccess}, nil
 	}
-	var rtStatus *metav1.Status
 	var pdbName string
 	err = func() error {
 		pdbs, err := r.getPodDisruptionBudgets(ctx, pod)
 		if err != nil {
 			return err
 		}
 
-		if len(pdbs) > 1 {
-			rtStatus = &metav1.Status{
-				Status:  metav1.StatusFailure,
-				Message: "This pod has more than one PodDisruptionBudget, which the eviction subresource does not support.",
-				Code:    500,
-			}
-			return nil
-		}
 		if len(pdbs) == 0 {
 			return nil
 		}
 
-		pdb := &pdbs[0]
-		pdbName = pdb.Name
 		refresh := false
-		err = retry.RetryOnConflict(EvictionsRetry, func() error {
-			if refresh {
-				pdb, err = r.podDisruptionBudgetClient.PodDisruptionBudgets(pod.Namespace).Get(pdbName, metav1.GetOptions{})
-				if err != nil {
-					return err
+		for _, withDryRun := range []bool{true, false} {
+			err = retry.RetryOnConflict(EvictionsRetry, func() error {
+				for i := range pdbs {
```

Comment:
Would this be better if we moved the retry logic inside the loop over PDBs? This means we will retry each PDB until we succeed instead of redoing all PDBs every time we get a conflict on either of them. It seems like the current approach could lead to unnecessary decrements on PDBs where there are no conflicts. 

Also, seems like the fix in checkAndDecrement to avoid decrementing the allowed disruptions multiple times for a single pod is no longer in this PR. Any reason why this is no longer here?

Reply:
I have put back the fix in checkAndDecrement to avoid decrementing the allowed disruptions multiple times (which would address the unnecessary decrements on PDBs where there are no conflicts).

Reply:
Any reason why you prefer the retry outside the loop over the PDBs? With the fix in checkAndDecrement I don't think it should have any impact on whether an eviction is allowed, but I think it could lead to unnecessary fetch and update for PDBs that have already been updated.

Reply:
'matching pdbs with one disruption allowed and one disruption disallowed' case wouldn't pass when the retry logic is inside the loop

Reply:
Why? Can you walk through the scenario that would fail?

Reply:
@mortent 
I have updated the PR and moved the retry inside.